src/dmd/dsymbol.d

```diff
@@ -362,6 +362,16 @@ extern (C++) class Dsymbol : ASTNode
         if (sc.isDeprecated())
             return false;
 
+        // Don't issue additional messages if this symbol is instantiated
+        // from a deprecated template
+        if (auto ti = this.isInstantiated())
+        {
+            assert(ti.tempdecl); // Should always be resolved here
+
+            if (ti.tempdecl.isDeprecated())
+                return false;
+        }
```

Comment:
Not happy about this but semantic analysis might only see the deprecated member without going through the template instance.